instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/SnsCamelTest.groovy

```diff
@@ -0,0 +1,696 @@
+            "aws.operation" "GetQueueAttributes"
+            "aws.queue.url" queueUrl
+            "aws.service" "AmazonSQS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(2, 1) {
+
+        span(0) {
+          name "SQS.SetQueueAttributes"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "SetQueueAttributes"
+            "aws.queue.url" queueUrl
+            "aws.service" "AmazonSQS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(3, 1) {
+
+        span(0) {
+          name "SNS.CreateTopic"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "CreateTopic"
+            "aws.service" "AmazonSNS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(4, 1) {
+
+        span(0) {
+          name "SNS.Subscribe"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "Subscribe"
+            "aws.service" "AmazonSNS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.transport" "IP.TCP"
+            "net.peer.port" {it == null || Number}
+          }
+        }
+      }
+      trace(5, 1) {
+        span(0) {
+          name "SQS.ReceiveMessage"
+          kind CLIENT
+          hasNoParent()
+          attributes {
+            "aws.agent" "java-aws-sdk"
+            "aws.endpoint" String
+            "aws.operation" "ReceiveMessage"
+            "aws.queue.url" queueUrl
+            "aws.service" "AmazonSQS"
+            "http.flavor" "1.1"
+            "http.method" "POST"
+            "http.status_code" 200
+            "http.url" String
+            "net.peer.name" String
+            "net.peer.port" {it == null || Number}
+            "net.transport" "IP.TCP"
+          }
+        }
+      }
+      trace(6, 3) {
+        span(0) {
+          name "SNS.Publish"
+          kind CLIENT
```

Comment:
Should this be PRODUCER?

Reply:
This is a tough question. 
In general, SNS message can be fan out to SQS (becoming a message), HTTP, lambda, kinesis or even email. In some of these cases we would have PUBLISH (send) - CLIENT (get) while now for SQS we have CLIENT (send) - CONSUMER (get). 
If we could, the best way IMO would be to modify "sending" span kind once a message is consumed (by SQS), but we can't. 
@anuraaga what do you think?

Reply:
From what I understand of the spec, we'd still generally expect the SNS message to be treated like messaging conventions so would be `PRODUCER`. The trickiness related to CLIENT and PRODUCER has come up before, I'm ok with just picking any for now but good chance of changing later.